<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SolverDialog</class>
 <widget class="QDialog" name="SolverDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>528</width>
    <height>739</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="minimumSize">
      <size>
       <width>510</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>MiniZinc</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>MiniZinc path</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="mznDistribPath"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="mznpath_select">
          <property name="text">
           <string>Select</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="check_solver">
          <property name="text">
           <string>Check</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="check_updates">
        <property name="text">
         <string>Check for updates to MiniZinc on startup once a day</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Found MiniZinc installation:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="mzn2fzn_version">
           <property name="text">
            <string>none</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Solver</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QComboBox" name="solvers_combo">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>1</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <item>
           <property name="text">
            <string>Add new...</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QFrame" name="solverFrame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Name</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="name"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Solver ID</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="solverId"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Version</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="version"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Executable</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="executable"/>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="exec_select">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Select</string>
             </property>
             <property name="autoDefault">
              <bool>true</bool>
             </property>
             <property name="flat">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="exeNotFoundLabel">
           <property name="text">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#fc0107;&quot;&gt;Executable not found!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="textFormat">
            <enum>Qt::RichText</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QCheckBox" name="detach">
             <property name="text">
              <string>Detach from IDE</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="needs_mzn2fzn">
             <property name="text">
              <string>Run with mzn2fzn</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="needs_solns2out">
             <property name="text">
              <string>Run with solns2out</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Solver library path</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="mznpath"/>
           </item>
           <item>
            <widget class="QPushButton" name="mznlib_select">
             <property name="text">
              <string>Select</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Supported standard command line flags:</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QCheckBox" name="has_stdflag_a">
             <property name="text">
              <string>-a</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_n">
             <property name="text">
              <string>-n</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_s">
             <property name="text">
              <string>-s</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_v">
             <property name="text">
              <string>-v</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_p">
             <property name="text">
              <string>-p</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_r">
             <property name="text">
              <string>-r</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_f">
             <property name="text">
              <string>-f</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="has_stdflag_t">
             <property name="text">
              <string>-t</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="requiredFlags">
        <property name="title">
         <string>Required solver flags</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="deleteButton">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="updateButton">
          <property name="text">
           <string>Update</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>mznDistribPath</tabstop>
  <tabstop>mznpath_select</tabstop>
  <tabstop>check_solver</tabstop>
  <tabstop>check_updates</tabstop>
  <tabstop>solvers_combo</tabstop>
  <tabstop>name</tabstop>
  <tabstop>solverId</tabstop>
  <tabstop>version</tabstop>
  <tabstop>executable</tabstop>
  <tabstop>exec_select</tabstop>
  <tabstop>detach</tabstop>
  <tabstop>needs_mzn2fzn</tabstop>
  <tabstop>needs_solns2out</tabstop>
  <tabstop>mznpath</tabstop>
  <tabstop>mznlib_select</tabstop>
  <tabstop>has_stdflag_a</tabstop>
  <tabstop>has_stdflag_n</tabstop>
  <tabstop>has_stdflag_s</tabstop>
  <tabstop>has_stdflag_v</tabstop>
  <tabstop>has_stdflag_p</tabstop>
  <tabstop>has_stdflag_r</tabstop>
  <tabstop>has_stdflag_f</tabstop>
  <tabstop>has_stdflag_t</tabstop>
  <tabstop>deleteButton</tabstop>
  <tabstop>updateButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SolverDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SolverDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
